src/components/InteractionLayer/index.js

```diff
@@ -6,10 +6,12 @@ import ScalerContext from '../../context/Scaler';
 import { createXScale, createYScale } from '../../utils/scale-helpers';
 import { seriesPropType, annotationPropType } from '../../utils/proptypes';
 import Annotation from '../Annotation';
+import Ruler from '../Ruler';
 
 class InteractionLayer extends React.Component {
   static propTypes = {
     crosshair: PropTypes.bool,
+    ruler: PropTypes.objectOf(PropTypes.func).isRequired,
```

Comment:
Could you use `shape` for this?
```
ruler: PropTypes.shape({xLabel: PropTypes.func.isRequired, yLabel: PropTypes.func.isRequired}),
```

Reply:
Already used that [here](https://github.com/cognitedata/griff-react/pull/52). I close this one in in favor of that PR.